### PR TITLE
vsql-test-only: test error paths

### PR DESCRIPTION
**Pull request by malone-at-work**
* Extension will be used to test edge cases and error paths
* Add a test to cause a decode failure - note that this test just describes current behavior
* Fix decode error that would return a garbage value

### CODE REVIEW

File: mysql-test/suite/villagesql/extension/r/extension_decode_failure.result [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+id	val
+1	OK hello
+2	OK world
+3	OK abc
+4	NULL
+Warnings:
+Warning	1366	Incorrect FAULT_BLOB value: 'DECODE_FAIL\x00\x00\x00\x00\x00' for column 'val' at row 4
+SHOW WARNINGS;
+Level	Code	Message
+Warning	1366	Incorrect FAULT_BLOB value: 'DECODE_FAIL\x00\x00\x00\x00\x00' for column 'val' at row 4
```
Comment (tomas-villagesql) [inline mysql-test/suite/villagesql/extension/r/extension_decode_failure.result:43]: is it possible to update a row with a faulty value? e.g. overwrite the faulty value

Comment (malone-at-work) [reply mysql-test/suite/villagesql/extension/r/extension_decode_failure.result:43]: Done.
